class/class5.1/MegaDataPool.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef MEGADATAPOOL_HPP
+#define MEGADATAPOOL_HPP
+
+#include <queue>
+#include <stdexcept>
+#include "MegaData.hpp"
+
+class MegaDataPool {
+public:
+    explicit MegaDataPool(size_t poolSize);
+
+    ~MegaDataPool();
+
+    MegaData* acquire();
```

Comment:
it is better to mark this method as nodiscard as we don't want this result to be ignored as it may cause a memory leak due to resource that will not be never returned.